libs/sdk-ui-ext/src/internal/components/configurationControls/comparison/calculation/CalculationControl.tsx

```diff
@@ -0,0 +1,54 @@
+// (C) 2023 GoodData Corporation
+import React, { useMemo } from "react";
+import { useIntl } from "react-intl";
+
+import { CalculationType } from "@gooddata/sdk-ui-charts";
+
+import { messages } from "../../../../../locales.js";
+import { calculationDropdownItems } from "../../../../constants/dropdowns.js";
+import DropdownControl from "../../DropdownControl.js";
+import CalculationListItem from "./CalculationListItem.js";
+
+import { COMPARISON_CALCULATION_TYPE_VALUE_PATH } from "../ComparisonValuePath.js";
+import { IComparisonControlProperties } from "../../../../interfaces/ControlProperties.js";
+import { IVisualizationProperties } from "../../../../interfaces/Visualization.js";
+
+interface ICalculationControlProps {
+    disabled: boolean;
+    properties: IVisualizationProperties<IComparisonControlProperties>;
+    pushData: (data: any) => any;
+}
+
+const CALCULATION_DROPDOWN_WIDTH = 194;
+
+const CalculationControl: React.FC<ICalculationControlProps> = ({ disabled, properties, pushData }) => {
+    const { formatMessage } = useIntl();
+    const calculationType: CalculationType = properties.controls?.comparison?.calculationType;
```

Comment:
Is there a default value in case it would be undefined?

Reply:
The default value of calculationType is based on the type of secondaryMeasure. Therefore, I've decided to handle it in PluggableHeadline instead.